packages/template-retail-react-app/app/hooks/use-shopper-context-search-params.test.js

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React from 'react'
+import {Router} from 'react-router'
+import {renderHook} from '@testing-library/react'
+import {createMemoryHistory} from 'history'
+import {
+    useShopperContextSearchParams,
+    getShopperContextFromSearchParams
+} from '@salesforce/retail-react-app/app/hooks/use-shopper-context-search-params'
+import {
+    SHOPPER_CONTEXT_FIELD_TYPES,
+    SHOPPER_CONTEXT_SEARCH_PARAMS
+} from '@salesforce/retail-react-app/app/constants'
+
+afterEach(() => {
+    jest.clearAllMocks()
+})
+
+describe('useShopperContextSearchParams', () => {
+    test('returns an empty object when no search params are present', () => {
+        const history = createMemoryHistory()
+        history.push('')
+
+        const wrapper = ({children}) => <Router history={history}>{children}</Router>
+        const {result} = renderHook(() => useShopperContextSearchParams(), {wrapper})
+        expect(result.current).toEqual({})
+    })
+
+    test('returns an empty object when search params not related to shopper context are present', () => {
+        const history = createMemoryHistory()
+        history.push('?a=1&b=2&c=3')
+
+        const wrapper = ({children}) => <Router history={history}>{children}</Router>
+        const {result} = renderHook(() => useShopperContextSearchParams(), {wrapper})
+        expect(result.current).toEqual({})
+    })
+
+    test('returns an object with parsed search params related to shopper context', () => {
```

Comment:
Please add a test for the scenario where there are shopper-context params that are _not_ allowed to go through.

For example:
`?a&b&c` → because `c` is not specified/allowed, only `a` and `b` can go through.